internal/pproflogging/pproflogging.go

```diff
@@ -97,10 +103,41 @@ var pprofProfileRates = map[ProfileName]pprofSetRate{
 	},
 }
 
+func newProfileConfigs(wrt Writer) *ProfileConfigs {
+	q := &ProfileConfigs{
+		wrt: wrt,
+	}
+
+	return q
+}
+
+// LoadProfileConfig configure PPROF profiling from the config in ppconfigss.
+func LoadProfileConfig(ctx context.Context, ppconfigss string) (map[ProfileName]*ProfileConfig, error) {
+	// if empty, then don't bother configuring but emit a log message - use might be expecting them to be configured
+	if ppconfigss == "" {
+		log(ctx).Debug("no profile configuration. skipping PPROF setup")
+		return nil, nil
+	}
+
+	bufSizeB := DefaultDebugProfileDumpBufferSizeB
+
+	// look for matching services.  "*" signals all services for profiling
+	log(ctx).Info("configuring profile buffers")
+
+	// acquire global lock when performing operations with global side-effects
+	return parseProfileConfigs(bufSizeB, ppconfigss)
+}
+
+// ProfileConfig configuration flags for a profile.
+type ProfileConfig struct {
+	flags []string
+	buf   *bytes.Buffer
+}
+
 // GetValue get the value of the named flag, `s`.  False will be returned
 // if the flag does not exist. True will be returned if flag exists without
 // a value.
-func (p ProfileConfig) GetValue(s string) (string, bool) {
+func (p *ProfileConfig) GetValue(s string) (string, bool) {
```

Comment:
Shouldn't this be a non-pointer  for receiver methods which are read-only ?

Reply:
Yes, it can be.  I wanted to stick with a "default" receiver type for the first implementation.  This code has never seen production, so I wanted to stick with some "least-common-denominator" implementations in the event that its not useful or needs to be changed.  I can make changes, if necessary, on a follow-on PR.

Reply:
lets make this simple change if you agree:

```suggestion
func (p ProfileConfig) GetValue(s string) (string, bool) {
```